File: src/main/java/uk/gov/justice/digital/hmpps/riskprofiler/model/OffenderSentenceTerms.java
package uk.gov.justice.digital.hmpps.riskprofiler.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.time.LocalDate;

@SuppressWarnings("unused")
@ApiModel(description = "Offender Sentence terms details for booking id")
@JsonInclude(JsonInclude.Include.NON_NULL)
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Data
public class OffenderSentenceTerms implements Serializable {
    @ApiModelProperty(required = true, value = "Offender booking id.", position = 1, example = "1132400")
    private Long bookingId;

    @ApiModelProperty(required = true, value = "Sentence number within booking id.", position = 2, example = "2")
    private Integer sentenceSequence;

    @ApiModelProperty(required = true, value = "Sentence term number within sentence.", position = 3, example = "1")
    private Integer termSequence;

    @ApiModelProperty(value = "Sentence number which this sentence follows if consecutive, otherwise concurrent.", position = 4, example = "2")
    private Integer consecutiveTo;

    @ApiModelProperty(value = "Sentence type, using reference data from table SENTENCE_CALC_TYPES.", position = 5, example = "2")
    private String sentenceType;

    @ApiModelProperty(value = "Sentence type description.", position = 6, example = "2")
    private String sentenceTypeDescription;

    @ApiModelProperty(required = true, value = "Start date of sentence.", position = 7, example = "2018-12-31")
    private LocalDate startDate;

    @ApiModelProperty(value = "Sentence length years.", position = 8)
    private Integer years;

    @ApiModelProperty(value = "Sentence length months.", position = 9)
    private Integer months;

    @ApiModelProperty(value = "Sentence length weeks.", position = 10)
    private Integer weeks;

    @ApiModelProperty(value = "Sentence length days.", position = 11)
    private Integer days;

    @ApiModelProperty(required = true, value = "Whether this is a life sentence.", position = 12)
    private Boolean lifeSentence;
}

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/model/OffenderSentenceTerms.kt
package uk.gov.justice.digital.hmpps.riskprofiler.model

import com.fasterxml.jackson.annotation.JsonInclude
import io.swagger.v3.oas.annotations.media.Schema
import java.io.Serializable
import java.time.LocalDate

@Schema(title = "Offender Sentence terms details for booking id")
@JsonInclude(JsonInclude.Include.NON_NULL)
data class OffenderSentenceTerms(
  @Schema(required = true, title = "Offender booking id.", example = "1132400")
  private val bookingId: Long? = null,

  @Schema(required = true, title = "Sentence number within booking id.", example = "2")
  private val sentenceSequence: Int? = null,

  @Schema(required = true, title = "Sentence term number within sentence.", example = "1")
  private val termSequence: Int? = null,

  @Schema(
    title = "Sentence number which this sentence follows if consecutive, otherwise concurrent.",
    example = "2",
  )
  private val consecutiveTo: Int? = null,

  @Schema(
    title = "Sentence type, using reference data from table SENTENCE_CALC_TYPES.",
    example = "2",
  )
  private val sentenceType: String? = null,

  @Schema(title = "Sentence type description.", example = "2")
  private val sentenceTypeDescription: String? = null,

  @Schema(required = true, title = "Start date of sentence.", example = "2018-12-31")
  private val startDate: LocalDate? = null,

  @Schema(title = "Sentence length years.")
  private val years: Int? = null,

  @Schema(title = "Sentence length months.")
  private val months: Int? = null,

  @Schema(title = "Sentence length weeks.")
  private val weeks: Int? = null,

  @Schema(title = "Sentence length days.")
  private val days: Int? = null,

  @Schema(required = true, title = "Whether this is a life sentence.")
  val lifeSentence: Boolean? = null,
) : Serializable {

  constructor(booking: Long, lifeSentence: Boolean) : this(bookingId = booking, lifeSentence = lifeSentence)
}
